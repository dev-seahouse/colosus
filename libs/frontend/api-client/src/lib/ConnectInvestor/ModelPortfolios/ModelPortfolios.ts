import ConnectInvestorBaseApi from '../_Base/Base';
import type { ConnectPortfolioSummaryDto } from '@bambu/shared';

export type ConnectInvestorGetModelPortfoliosSummaryResponseDto =
  ConnectPortfolioSummaryDto.IConnectPortfolioSummaryDto[];

export class ConnectInvestorAdvisorModelPortfoliosApi extends ConnectInvestorBaseApi {
  constructor(private readonly apiPath = '/model-portfolios') {
    super();
  }

  /**
   * Gets model portfolio summary.
   * - {@link http://localhost:9000/openapi#/Connect%20Investor/ConnectInvestorController_GetInvestorModelPortfolioSummary}.
   */
  public async getModelPortfoliosSummary() {
    return await this.axios.get<ConnectInvestorGetModelPortfoliosSummaryResponseDto>(
      `${this.apiPath}/summary`
    );
  }
}

export default ConnectInvestorAdvisorModelPortfoliosApi;
