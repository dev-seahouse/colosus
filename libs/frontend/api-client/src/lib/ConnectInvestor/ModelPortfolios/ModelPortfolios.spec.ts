import { describe } from 'vitest';
import { ConnectInvestorAdvisorModelPortfoliosApi } from './ModelPortfolios';

describe('AdvisorProfile', () => {
  const connectInvestorAdvisorProfileApi =
    new ConnectInvestorAdvisorModelPortfoliosApi();

  describe('getModelPortfoliosSummary', () => {
    it('should return correct response', async () => {
      const res =
        await connectInvestorAdvisorProfileApi.getModelPortfoliosSummary();
      expect(res.data).toMatchSnapshot();
    });
  });
});
