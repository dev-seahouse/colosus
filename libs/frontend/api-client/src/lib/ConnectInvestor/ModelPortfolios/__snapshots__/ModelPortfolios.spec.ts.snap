// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdvisorProfile > getModelPortfoliosSummary > should return correct response 1`] = `
[
  {
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "0",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "30",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "70",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors who want to preserve their capital, who are OK with low returns and/or who are investing with a short term time horizon.",
    "expectedReturnPercent": "3",
    "expectedVolatilityPercent": "4",
    "id": "f10502f5-6f15-4eb1-9524-dc09bafceec7",
    "key": "CONSERVATIVE",
    "name": "Conservative Portfolio",
    "reviewed": true,
    "riskProfileId": "8e6fb5fd-e8b4-4019-9d2c-6fd1ff8f6e5f",
    "showSummaryStatistics": true,
  },
  {
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "25",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "20",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "55",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors who are willing to take on a little bit of risk to increase marginally their returns. It has a mid to long-term investment time horizon.",
    "expectedReturnPercent": "4",
    "expectedVolatilityPercent": "7.5",
    "id": "dc59fcd3-0a84-4726-a2ad-f07ce20f367a",
    "key": "MODERATE",
    "name": "Moderate Portfolio",
    "reviewed": false,
    "riskProfileId": "eeef55b0-40a7-4980-b8ec-a94e01be1d13",
    "showSummaryStatistics": true,
  },
  {
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "50",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "10",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "40",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors who are seeking average returns and are ready to tolerate some price fluctuations. It has a mid to long-term investment time horizon.",
    "expectedReturnPercent": "5",
    "expectedVolatilityPercent": "11",
    "id": "bca96485-8dc2-41fc-aabf-2a3f4745aab8",
    "key": "BALANCED",
    "name": "Balanced Portfolio",
    "reviewed": false,
    "riskProfileId": "4bf6874c-e5ea-4f66-b2f9-e1212b823422",
    "showSummaryStatistics": true,
  },
  {
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "75",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "5",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "20",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors seeking growth and who are willing to take on higher risk. It has a long-term investment time horizon.",
    "expectedReturnPercent": "6.5",
    "expectedVolatilityPercent": "14.5",
    "id": "62f081f8-5220-441d-b262-c050da78602e",
    "key": "GROWTH",
    "name": "Growth Portfolio",
    "reviewed": false,
    "riskProfileId": "e39f6a89-743f-4ed2-9307-22d5d545020c",
    "showSummaryStatistics": true,
  },
  {
    "assetClassAllocation": [
      {
        "assetClass": "Equity",
        "included": true,
        "percentOfPortfolio": "100",
      },
      {
        "assetClass": "Money Market",
        "included": true,
        "percentOfPortfolio": "0",
      },
      {
        "assetClass": "Bonds",
        "included": true,
        "percentOfPortfolio": "0",
      },
      {
        "assetClass": "Other",
        "included": true,
        "percentOfPortfolio": "0",
      },
    ],
    "description": "This portfolio is suitable for investors seeking high returns and who are willing to take on high risk. It has a long-term investment time horizon.",
    "expectedReturnPercent": "8",
    "expectedVolatilityPercent": "18",
    "id": "5aea8dfa-e6ed-467f-9541-34b19884ad8d",
    "key": "AGGRESSIVE",
    "name": "Aggressive Portfolio",
    "reviewed": false,
    "riskProfileId": "149a15e6-7152-4a2d-9d16-0d6a0c70b825",
    "showSummaryStatistics": true,
  },
]
`;
